include/aws/iotdevice/secure_tunneling.h

```diff
@@ -6,120 +6,260 @@
-
-AWS_IOTDEVICE_API
-int aws_secure_tunnel_close(struct aws_secure_tunnel *secure_tunnel);
-
-AWS_IOTDEVICE_API
-int aws_secure_tunnel_send_data(struct aws_secure_tunnel *secure_tunnel, const struct aws_byte_cursor *data);
-
-AWS_IOTDEVICE_API
-int aws_secure_tunnel_stream_start(struct aws_secure_tunnel *secure_tunnel);
-
-AWS_IOTDEVICE_API
-int aws_secure_tunnel_stream_reset(struct aws_secure_tunnel *secure_tunnel);
-
 /**
- * Raises exception and returns AWS_OP_ERR if options are missing required parameters.
+ * Asynchronous notify to the secure tunnel that you want it to attempt to connect.
+ * The secure tunnel will attempt to stay connected.
+ *
+ * @param secure_tunnel secure tunnel to start
+ * @return success/failure in the synchronous logic that kicks off the start process
  */
 AWS_IOTDEVICE_API
-int aws_secure_tunnel_options_validate(const struct aws_secure_tunnel_options *options);
+int aws_secure_tunnel_start(struct aws_secure_tunnel *secure_tunnel);
 
 /**
- * Create persistent storage for aws_secure_tunnel_options.
- * Makes a deep copy of (or acquires reference to) any data referenced by options,
+ * Asynchronous notify to the secure tunnel that you want it to transition to the stopped state. When the
+ * secure tunnel reaches the stopped state, all session state is erased.
+ *
+ * @param secure_tunnel secure tunnel to stop
+ * @return success/failure in the synchronous logic that kicks off the start process
  */
 AWS_IOTDEVICE_API
-struct aws_secure_tunnel_options_storage *aws_secure_tunnel_options_storage_new(
-    const struct aws_secure_tunnel_options *options);
+int aws_secure_tunnel_stop(struct aws_secure_tunnel *secure_tunnel);
 
 /**
- * Destroy options storage, and release any references held.
+ * Queues a message operation in a secure tunnel
+ *
+ * @param secure_tunnel secure tunnel to queue a message for
+ * @param message_options configuration options for the message operation
+ * @return success/failure in the synchronous logic that kicks off the message operation
  */
 AWS_IOTDEVICE_API
-void aws_secure_tunnel_options_storage_destroy(struct aws_secure_tunnel_options_storage *storage);
+int aws_secure_tunnel_send_message(
+    struct aws_secure_tunnel *secure_tunnel,
+    const struct aws_secure_tunnel_message_view *message_options);
 
-/**
- * Return pointer to options struct stored within.
- */
 AWS_IOTDEVICE_API
-const struct aws_secure_tunnel_options *aws_secure_tunnel_options_storage_get(
-    const struct aws_secure_tunnel_options_storage *storage);
+int aws_secure_tunnel_stream_reset(
+    struct aws_secure_tunnel *secure_tunnel,
+    const struct aws_secure_tunnel_message_view *message_options);
 
-/* Making this exposed public to verify testing in the sdk layer */
+//***********************************************************************************************************************
+/* THESE API SHOULD ONLY BE USED FROM SOURCE MODE */
```

Comment:
This is a weird area because TECHNICALLY we are only responsible for the secure tunnel as a Destination device but since we've already exposed API that allows source mode operations I didn't feel comfortable removing it completely from access to prevent breaking existing API. If we decide it's okay to remove the ability to start streams and function at all as a source device, I can definitely move these into private for testing only. For now I'm planning on labeling this functionality as deprecated but I'm not 100% on how I want to communicate this change yet.